dotCMS/src/main/java/com/dotcms/rest/api/v1/system/role/RoleResource.java

```diff
@@ -214,4 +226,72 @@ public Response findRoleLayouts(
 				layoutAPI.loadLayoutsForRole(role)
 		)).build();
 	}
+
+	/**
+	 * Load the user and roles by role id.
+	 * @param request   {@link HttpServletRequest}
+	 * @param response  {@link HttpServletResponse}
+	 * @param roleId    {@link String} role
+	 * @param roleHierarchyForAssign {@link Boolean} true if want to include the hierarchy, false by default
+	 * @param roleNameToFilter {@link String} prefix role name, if you want to filter the results
+	 * @return Response
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 */
+	@GET
+	@Path("/{roleid}/rolehierarchyanduserroles")
+	@Produces("application/json")
+	@SuppressWarnings("unchecked")
+	public Response loadUsersAndRolesByRoleId(@Context final HttpServletRequest request,
+			@Context final HttpServletResponse response,
+			@PathParam   ("roleid") final String roleId,
+			@DefaultValue("false") @QueryParam("roleHierarchyForAssign") final boolean roleHierarchyForAssign,
+			@QueryParam  ("name") final String roleNameToFilter) throws DotDataException, DotSecurityException {
+
+		new WebResource.InitBuilder(this.webResource).requiredBackendUser(true)
+				.requiredFrontendUser(false).requestAndResponse(request, response)
+				.rejectWhenNoUser(true).init();
+
+		final Role role = this.roleAPI.loadRoleById(roleId);
+
+		if (null == role || !UtilMethods.isSet(role.getId())) {
+
+			throw new DoesNotExistException("The role: " + roleId + " does not exists");
+		}
+
+		final List<Role> roleList = new ArrayList<>();
+		final List<User> userList = new ArrayList<>();
+
+		Logger.debug(this, ()->"loading users and roles by role: " + roleId);
+
+		if (!role.isUser()) {
+
+			userList.addAll(this.roleAPI.findUsersForRole(role, roleHierarchyForAssign));
+			roleList.addAll(roleHierarchyForAssign? this.roleAPI.findRoleHierarchy(role): Arrays.asList(role));
+		} else {
+
+			userList.add(this.userAPI.loadUserById(role.getRoleKey(), APILocator.systemUser(), false));
+		}
+
+		for (final User user : userList) {
+
+			final Role roleToTest = this.roleAPI.getUserRole(user);
+			if (roleToTest != null && UtilMethods.isSet(roleToTest.getId())) {
+
+				roleList.add(roleToTest);
+			}
+		}
+
+		return Response.ok(new ResponseEntityView(
+				null != roleNameToFilter? this.filterRoleList(roleNameToFilter, roleList):roleList)).build();
+	}
+
+	private final List<Role> filterRoleList(final String roleNameToFilter, final List<Role> roleList) {
```

Comment:
Codacy found an issue: [Unnecessary modifier 'final' on method 'filterRoleList': private methods cannot be overridden](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5919676)